### DISCUSSION TITLE
before_close event

### DISCUSSION THREAD

**Discussion by Offerel** [biati-digital/glightbox#499] [Q&A] [answered]
I would like to ask, if there is a before_close event or if not, if there is a way for some sort of workaround?

Reason is, i have added some custom buttons and i use the close button, to position these buttons next to each other. If i now close the picture, the buttons disappear also, but only after some time. This means, a gallery with white background will display the buttons for a blink of a second, between the image is closed and the buttons are completely fade out. This is not the a big issue, only a small disadvantage.

**Comment by gingerchew** [comment, marked as answer]
No but you could add an event listener in the `open` event to the close button, then you can do any cleanup work you need to do in that handler.

```js
lightbox.on('open', () => {
	document.querySelector({{ the glightbox close button selector }})
		.addEventListener('click', () => {
			// do cleanup stuff
		}, { once: true })
})
```

**Comment by Offerel** [comment]
Many thx. Works perfectly. 

BTW: I wasn't aware, that there is a `{ once: true })` for the addEventListener function.

For reference, im using it in https://codeberg.org/Offerel/roundcube_pictures. This is a photo plugin for Roundcube.

**Reply by gingerchew** [reply]
Yeah, it was added a while ago, but it doesn't show up in many tutorials. Removing stuff like this is really nice!

```js
document.addEventListener('click', function runClickOnce() {
	// ...stuff...
	document.removeEventListener('click', runClickOnce);
});
```
